### PR TITLE
Mistral 7B Instruct with ChatCompletions interface

### PR DESCRIPTION
**Pull request by joostinyi**

### CODE REVIEW

**Review by squidarth — COMMENTED** [review]
awesome, thanks for working on this @joostinyi! Couple things:

* Instead of returning `{"text": ...}`, could we just return the string?
* Does this include the prompt in the output? I had to do some finagling to make that not happen

**Comment by joostinyi** [comment]
@squidarth I can make the change to not wrap the non-streaming output. In the same vein, TRT-LLM trusses use streaming mode by default, which returns just the output tokens – however if we pass `stream: false` it will return the prompt + the output generation. Is the desired behavior such that we don't show the prompt in the non-streaming case?

**Comment by squidarth** [comment]
>  Is the desired behavior such that we don't show the prompt in the non-streaming case?

Yeah, that's right. With the transformers library, there's a `return_full_text` parameter that I set to `false`

**Comment by joostinyi** [comment]
@aspctu do you know if the `TritonClient` is amenable to this behavior?

> Yeah, that's right. With the transformers library, there's a `return_full_text` parameter that I set to `false`

**Comment by squidarth** [comment]
It's not a huge deal if it's not possible

**Comment by joostinyi** [comment]
We'll come back to this @squidarth

**Comment by ttim** [comment]
Could we keep it as templated configuration as before and either:
- Add `model.py` as ignored file so it can to be different there or
- Change template and make `openai-api` behavior optional in core truss, i.e. if `model_metadata.open_ai_api` is `True` then do what you're doing otherwise do old things

I think second option is preferable

**Comment by squidarth** [comment]
> Let's merge as is and come back to this if that's okay @squidarth

Yeah, no rush, it's a nice-to-have

**Review by aspctu — APPROVED** [review]

File: templates/trt-llm/config.yaml [hunk 1]
```diff
@@ -2,13 +2,14 @@ base_image:
   image: nvcr.io/nvidia/tritonserver:23.10-trtllm-python-py3
   python_executable_path: /usr/bin/python3
 model_metadata:
-  example_model_input: {"prompt": "What's the meaning of life?", "max_tokens": 1024}
+  example_model_input: {}
```
[the remark this answers is not in the document]
Comment (joostinyi) [reply templates/trt-llm/config.yaml:5]: hack around one line json in the `generate.py` expects this key

File: templates/trt-llm/config.yaml [hunk 2]
```diff
@@ -2,13 +2,14 @@ base_image:
...
   tags:
   - text-generation
   tensor_parallelism: 1
 python_version: py311
 requirements:
 - tritonclient[all]
+- transformers==4.34.1
```
[the remark this answers is not in the document]
Comment (joostinyi) [reply templates/trt-llm/config.yaml]: i've refactored to only the mistral config – this is from [upstream](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1#troubleshooting)

File: llama/llama-2-7b-trt-llm/model/model.py [hunk 3]
```diff
@@ -52,17 +53,24 @@ def load(self):
...
-        self.eos_token_id = tokenizer.eos_token_id
+        self.eos_token_id = self.tokenizer.eos_token_id
 
     def predict(self, model_input):
         user_data = UserData()
         model_name = "ensemble"
         stream_uuid = str(next(self._request_id_counter))
 
-        prompt = model_input.get("prompt")
+        if "openai-compatible" in self._config["model_metadata"]["tags"]:
```
[the remark this answers is not in the document]
Comment (joostinyi) [reply llama/llama-2-7b-trt-llm/model/model.py]: Sure we can take this in a follow up

**Pull request merged**